Library/Homebrew/cmd/update.sh

```diff
@@ -336,23 +349,41 @@ EOS
         git fetch --force "${QUIET_ARGS[@]}" origin \
           "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" 2>/dev/null
       else
-        git fetch --force "${QUIET_ARGS[@]}" origin \
-          "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" || \
-            odie "Fetching $DIR failed!"
+        if ! git fetch --force "${QUIET_ARGS[@]}" origin \
+          "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH"
+        then
+          echo "Fetching $DIR failed!" >> "$HOMEBREW_UPDATE_FAILED_FILE"
+        fi
       fi
     ) &
   done
 
   wait
   trap - SIGINT
 
+  if [[ -f "$HOMEBREW_UPDATE_FAILED_FILE" ]]
+  then
+    onoe < "$HOMEBREW_UPDATE_FAILED_FILE"
+    rm -f "$HOMEBREW_UPDATE_FAILED_FILE"
+    export HOMEBREW_UPDATE_FAILED="1"
+  fi
+
+  unset HOMEBREW_UPDATED
+
   for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
   do
     [[ -d "$DIR/.git" ]] || continue
     pull "$DIR"
   done
 
   chdir "$HOMEBREW_REPOSITORY"
-  brew update-report "$@"
-  return $?
+
+  if [[ -n "$HOMEBREW_UPDATED" ]]
```

Comment:
Shouldn't this be `if [[ -n "$HOMEBREW_UPDATED" || -n "$HOMEBREW_UPDATE_FAILED" ]]` to get the correct error behavior? Otherwise, e.g., failing to fetch updates for all repositories will print the error, but still exit with a zero exit status.